src/coreclr/src/pal/inc/pal_mstypes.h

```diff
@@ -213,8 +213,6 @@ extern "C" {
 // includes are not included, so we need to define them.
 #ifndef PAL_IMPLEMENTATION
 
-// OS X already defines these types in 64 bit
-#if !defined(TARGET_OSX)
```

Comment:
The PR uncovered an interesting issue: before `TARGET_OSX` was already set in src/coreclr/src/pal/src/CMakeLists.txt and used here, but _not_ set when accessed from other object files.

Now that `TARGET_OSX` is defined globally this code here got removed which resulted in a compilation error:

```
In file included from /Users/alexander/dev/runtime/src/coreclr/src/nativeresources/resourcestring.cpp:7:
In file included from /Users/alexander/dev/runtime/src/coreclr/src/inc/utilcode.h:25:
/Users/alexander/dev/runtime/src/coreclr/src/inc/daccess.h:2350:14: error: unknown type name 'uint32_t'
typedef DPTR(uint32_t) PTR_uint32_t;
             ^
```

Removing the #if seems to work fine but I'd appreciate a second set of eyes. @jkotas

Reply:
It is fine since it is under #ifndef PAL_IMPLEMENTATION and so it can never collide with platform header files. We used to build part of PAL, the safecrt, without this define in the past, but it was fixed about a week ago in #32800, so the define doesn't make sense here anyways.